File: C2108G1/ADF1/06-09-2022/myapp/src/main/java/org/myapp/nguyenvana/models/Result.java
package org.myapp.nguyenvana.models;
public class Result {
    //init function = constructor
    public Result(float math, float physics, float chemistry) {
        this.math = math;
        this.physics = physics;
        this.chemistry = chemistry;
    }

    //fields
    private float math;
    private float physics;
    private float chemistry;

    public void showDetail(){
        System.out.println(String.format("math = %f \n physics = %f,\n chemistry = %f"
                ,this.math, this.physics, this.chemistry));
    }

    @Override
    public String toString() {
        return "Result' details is: {" +
                "math=" + math +
                ", physics=" + physics +
                ", chemistry=" + chemistry +
                '}';
    }
}

File: C2110I/AP/09-11-2022/myapp/myapp/models/Result.cs
using System;
//namespace = "package in Java"
namespace myapp
{
	public class Result
	{
        //private float math;//fields/atribute 
        //private float physics;
        //private float chemistry;
        //getter,

        public float Math { get; set; } //property, short-form
        public float Physics { get; set; }
        public float Chemistry { get; set; }

        //property is "Function"

        //full-form
        //private float math;//fields/atribute 
        //public float Math {
        //    get {
        //        Console.WriteLine("get get");
        //        return math;
        //    }
        //    set {
        //        Console.WriteLine("call SET");//validate data here !
        //        math = value;
        //    }
        //}
        //get detail object's information => override method ToString()
        public override string ToString()
        {
            return  $"Math = {Math}, " +
                    $"Physics = {Physics}, " +
                    $"Chemistry = {Chemistry}";
        }
    }
}

